src/sinks/datadog/events/sink.rs

```diff
@@ -52,7 +48,9 @@ where
     }
 }
 
-async fn ensure_required_fields(mut log: LogEvent) -> Option<LogEvent> {
+async fn ensure_required_fields(event: Event) -> Option<Event> {
```

Comment:
Because if we didn't do it here, we'd need to wrap it into an `Event` again to run the `Transformer` and `Encoder`, and would need to add a dedicated `impl Encoder<LogEvent> for (Transformer, crate::codecs::Encoder<()>)`, very similar to

https://github.com/vectordotdev/vector/blob/38a83a0e0f4952e5479e8966e0d83e663701339f/src/sinks/util/encoding.rs#L56-L67

Reply:
Ah, right, the transformer. Gotcha. 👍🏻

Reply:
Given that the `Transformer` only operates on `logs` internally, we could revisit if it would make sense for it to accepts logs only and change the signature accordingly.

I'll leave that out of scope for now, though